### DISCUSSION TITLE
Position dependant detection

### DISCUSSION THREAD

**Discussion by Gregorino** [ultralytics/yolov5#6675] [Q&A]
I have a yolo v5 large trained for custom object (ball) on a custom dataset.

When doing some analysis of the detection performance I found the following which I don't understand.

If I create a synthetic image (fully black) and paste my object over it at different locations (one at a time), the trained model doesn't detect the objects on all positions (it appears it has some blind spots). The location of this blind spots varies with the background color.

Probably this is due to training dataset where object was apparently mostly present only on a specific backgrounds,

BUT

shouldn't yolo behave the same for all image positions due to its construction if the background color is constant?

**Comment by glenn-jocher** [comment]
@Gregorino 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. See https://github.com/ultralytics/yolov5/issues/528 for information on grid sensitivity in YOLOv3. I have not had time to reproduce this experiment with YOLOv5.

We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃
